pkg/api/common/versions.go

```diff
@@ -198,37 +198,99 @@ var AllKubernetesSupportedVersions = map[string]bool{
 	"1.19.0-beta.0":  true,
 }
 
+// AllKubernetesSupportedVersionsOnAzureStack is a whitelist map of all supported Kubernetes version strings on Azure Stack
+// The bool value indicates if creating new clusters with this version is allowed
+var AllKubernetesSupportedVersionsOnAzureStack = map[string]bool{
+	"1.14.7":  false,
+	"1.14.8":  false, // disabled because of https://github.com/Azure/aks-engine/issues/2312
+	"1.15.4":  false,
+	"1.15.5":  false, // disabled because of https://github.com/Azure/aks-engine/issues/2312
+	"1.15.10": false,
+	"1.15.11": true,
+	"1.15.12": true,
+	"1.16.8":  false,
+	"1.16.9":  true,
+	"1.16.10": false,
+	"1.17.4":  false,
+	"1.17.5":  true,
+	"1.17.6":  false,
+}
+
+// AllKubernetesWindowsSupportedVersionsOnAzureStack maintain a set of available k8s Windows versions in aks-engine on Azure Stack
+var AllKubernetesWindowsSupportedVersionsOnAzureStack = map[string]bool{
+	"1.15.10": false,
+	"1.15.11": true,
+	"1.15.12": true,
+	"1.16.8":  false,
```

Comment:
Why `1.16.8`? do not think we ever supported that one

Reply:
The way I determine the supported versions is based on the Windows k8s binary URLs (https://github.com/Azure/aks-engine/blob/master/vhd/packer/configure-windows-vhd.ps1#L86-L94) and Linux k8s binaries (https://github.com/Azure/aks-engine/blob/master/vhd/packer/install-dependencies.sh#L349-L364). That may not be the best way, I will update.